src/components/Admin/QuestionDisplay.js

```diff
@@ -0,0 +1,212 @@
+import React, { Component, Fragment } from "react";
+import styled from "styled-components";
+
+import Button from "react-bootstrap/Button";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+
+import { withFirebase } from "upe-react-components";
+
+import Loader from "../Loader";
+import EditQuestion from "./EditQuestion";
+import { Container } from "../../styles/global";
+
+const StyledCol = styled(Col)`
+  padding: 10px 10px 10px 10px;
+`;
+
+const StyledHr = styled.hr`
+  border: 2px solid #333;
+  border-radius: 5px;
+`;
+
+const StyledDiv = styled.div`
+  width: 90%;
+  padding: 15px;
+  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
+  border-radius: 15px;
+  text-align: center;
+  &:hover {
+    -webkit-transform: translateY(-5px);
+    transform: translateY(-5px);
+    transition: all 0.3s linear;
+  }
+
+  a {
+    color: white;
+    font-weight: bold;
+    padding-top: 10px;
+    padding-bottom: 10px;
+    border-bottom: 1px solid grey;
+  }
+
+  a[aria-current="page"] {
+    color: ${(props) => props.theme.palette.mainBrand};
+  }
+`;
+
+class QuestionDisplay extends Component {
+  constructor(props) {
+    super(props);
+
+    this.toggleEdit = this.toggleEdit.bind(this);
+    this.toggleDelete = this.toggleDelete.bind(this);
+    this.updateData = this.updateData.bind(this);
+  }
+
+  _initFirebase = false;
+  state = {
+    loading: true,
+    error: null,
+    question: null,
+    editQuestion: false,
+    deleteQuestion: false,
+  };
+  unsub = null;
+
+  componentDidMount() {
+    if (this.props.firebase && !this._initFirebase) this.loadSettings();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.firebase && !this._initFirebase) this.loadSettings();
+  }
+
+  componentWillUnmount() {
+    if (typeof this.unsub === "function") this.unsub();
+  }
+
+  updatePage = () => {
+    this.loadSettings();
+  };
+
+  loadSettings = async () => {
+    this._initFirebase = true;
+
+    this.setState({ question: this.props.question, loading: false });
+  };
+
+  toggleEdit = () => {
+    this.setState({ editQuestion: !this.state.editQuestion });
+  };
+
+  toggleDelete = () => {
+    this.setState({ deleteQuestion: !this.state.deleteQuestion });
+  };
+
+  deleteQ = () => {
```

Comment:
Can we rename this to `deleteQuestion` and move this to parent component, so it's passed down like updateFunc. This component should only display data from the parent, and any changes to data should be handled by the parent (so no tracking state inside this component, since it's provided as props).

Reply:
Resolved in commit 21706d9246057ef3da00d313a3a1ea132dedb6c1